radiography/src/main/java/radiography/TreeStringRenderer.kt

```diff
@@ -3,12 +3,7 @@ package radiography
 internal abstract class TreeStringRenderer<N> {
 
   abstract fun StringBuilder.renderNode(node: N)
-
-  abstract fun N.getChildAt(index: Int): N?
-
-  abstract val N.childCount: Int
-
-  abstract fun N.matches(): Boolean
+  abstract val N.children: Collection<N?>
```

Comment:
Should we change this to `Collection<N>` ? ie not nullable?

Reply:
was nullable because of getChildAt and concurrent changes. Now the impl can take care of this.

Reply:
yea, that's better